code/modules/antagonists/cult/cult_items.dm

```diff
@@ -55,6 +65,26 @@ Striking a noncultist, however, will tear their flesh."}
 	else
 		return FALSE
 
+/obj/item/melee/cultblade/dagger/on_thrown(mob/living/carbon/user, atom/target)
+	. = ..()
+	if(!.)
+		return
+	thrower = user
```

Comment:
You don't need to independently track thrower, you are (or should be) able to get `thrower` from the throwing datum